tests/visual/scenes/filters/clip-to-viewport.scene.ts

```diff
@@ -0,0 +1,227 @@
+import { Assets } from '../../../../src/assets/Assets';
+import { Filter } from '../../../../src/filters/Filter';
+import { Sprite } from '../../../../src/scene/sprite/Sprite';
+
+import type { Container } from '../../../../src/scene/container/Container';
+import type { TestScene } from '../../types';
+
+export const scene: TestScene = {
+    it: 'clip to viewport filter texture',
+    only: true,
+    create: async (scene: Container) =>
+    {
+        const texture = await Assets.load('desert.jpg');
+        const sprite = new Sprite({
+            texture,
+            width: 128,
+            height: 128,
+        });
+
+        sprite.x = -64;
+
+        const wgslFragment = `
+struct GlobalFilterUniforms {
```

Comment:
Those shaders are duplicated between 2 tests, will move them to own fiels and share between tests.